src/types.ts

```diff
@@ -251,28 +251,28 @@ export function notDefaultCheck(ctx: Context, field: FieldDescriptorProto, place
       const zerothValue = enumProto.value.find((v) => v.number === 0) || enumProto.value[0];
       if (options.stringEnums) {
         const enumType = messageToTypeName(ctx, field.typeName);
-        return code`${place} !== ${enumType}.${zerothValue.name}`;
+        return code`${place} !== undefined && ${place} !== ${enumType}.${zerothValue.name}`;
```

Comment:
Hm, maybe we could do:

```
const maybeNotUndefined = options.useOptionals ? `${place} !== undefined && ` : "";
```

At the start of this method, and then in these places do:

```
return code`${maybeNotUndefined} ${place} !== ${enumType}.${zerothValue.name}`;
```

So that users that don't have `useOptional`s set won't have the unnecessary `!== undefined` in their output.

Granted, it doesn't hurt, but I think I'd like to keep it out if possible.

Reply:
I actually had problems with this before implementing `force/useOptionals`. Suppose that you have a message FooRequest with a `repeated int64 list = 0;` (remember you can't mark repeateds as optional), and an RPC `SendFoo(FooRequest)`. This code, intended to send an empty list (as with Go semantics), would trigger an error:

```
SendFoo({})
```

The error triggered is quite deep inside ts-proto and reads something like "list is not iterable", regardless of the value of force/useOptionals. IMHO, it is a small price to pay to always do the undefined check to allow this code to work normally.

If you strongly prefer to only enable this behaviour with force/useOptionals enabled then I can apply your suggestion, but personally I would strongly prefer not to.

Reply:
OK, I'll revert to only add the undefined-check if `useOptionals` is set and make a separate issue for the `[]` being undefined in a partial object, so we can discuss it further there. :)